components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -182,6 +195,24 @@ private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse() {
         return userAttributes;
     }
 
+    private void resolveGroupsForFederatedUser(Map<ClaimMapping, String> claimMappings) {
+
+        if (userFromResponse.getUser().getGroups() != null) {
+            claimMappings.put(buildClaimMapping(GROUP_CLAIM), String.join(
+                    FrameworkUtils.getMultiAttributeSeparator(), userFromResponse.getUser().getGroups()));
```

Comment:
Ideally we should validate the groups coming from the external service against this separator value